### ISSUE TITLE
Album and Folder Not Updating Correctly

### ISSUE THREAD

**Issue by mikekenyon99** [RhetTbull/osxphotos#1420]
I used the option to export into Folders my Photo Albums: 

`--directory "{folder_album}"`

I ran this option and because I'm saving to a OneDrive folder on my Mac it didn't like the fact one of my albums/folder had a **/** in it.  It generates this error, note that it puts. colon (:) not the actual offending character:

<img width="358" alt="image" src="https://github.com/RhetTbull/osxphotos/assets/42475982/c6d97e73-d04b-4981-a435-0f915c3154be">

No problem.  I left it to complete and then renamed the folder in Photos and let it sync around iCloud.  As you can see here:

<img width="284" alt="Screenshot 2024-02-28 at 07 43 50" src="https://github.com/RhetTbull/osxphotos/assets/42475982/4dd9f5fc-359e-4046-959f-5fb7eeb7559e">

I have then re-run osxphotos and it regenerates the old folder name!  I.e. it's not looking at Photos and then saying "_ah I need to change or rexport that_".

<img width="309" alt="Screenshot 2024-02-28 at 07 38 28" src="https://github.com/RhetTbull/osxphotos/assets/42475982/19ca6096-3f67-4087-81fc-48c384199f5a">

**Desktop (please complete the following information):**
 - OS: Sonoma 14.3.1
 - osxphotos version: version 0.67.3

Basically, I am stuck now, whatever I do OSXPhotos fails to update the folder name. 
Hope someone can help!

Mike

**Comment by oPromessa** [comment]
- So to confirm. You've updated the album name by removing : and / within Photos. And have relaunched osxphotos and still the folder name is wrong. Hmmm weird. 

- try simply deleting the destination folder and reexport to force osxphotos to go thru it again. 

- on the --directory option you can also convert characters with the magical template system (do read about it on the doc as it has a lot to offer!) I use album and convert / and : to . "{album[/,.|:,.]}"

**Comment by mikekenyon99** [comment]
Your first bullet point is correct - that is what is happening.

I did try what you said - I deleted the folder, and then re-exported but it has come back.  It's almost like OSXPhotos is caching it.  However, your suggestion of the magical template system is a good one.  Though that feels like a work-around not a solution. 

Much appreciated 👍

**Comment by RhetTbull** [comment]
I just tried to re-create this and could not. Here's what I did:

Create an album with a "/" in the name: "Test/Album"
Export photos
Album is created with a "/" in the name and photo exported: "Test/Album"
Deleted the folder "Test/Album" in Finder
Replaced the "/" with "-" in Photos: "Test-Photo"
Exported again with --update
Photos in the album were now exported to "Test-Album". The original album was not re-created.

>  It's almost like OSXPhotos is caching it.

osxphotos does not cache the album name. It does keep a record of every file exported in the export database (`.osxphotos_export.db` in the export folder) but this is used to match exported photos to the correct photo in Photos. Information such as the correct folder name is created at the time of export from the data in Photos, not the export database.

Are you sure there isn't a second album with the uncorrected name in Photos?

>  It generates this error, note that it puts. colon (:) not the actual offending character:

Note that it's actually osxphotos not OneDrive that put a colon in instead of a slash. In the macOS filesystem "/" is used to separate folders and while it's possible to create a file with a "/" in it, this is ill-advised and will make the file very difficult to work with using normal command line tools. osxphotos instead changes all "/" in a file name to ":".  The Finder will automatically display these as "/" even though on disk, they're actually stored as ":" (this is a throwback to macOS Classic which used ":" to separate folders).

**Comment by mikekenyon99** [comment]
Hi - thanks for update.  In that case I'm not sure what to do then.   I'll double check my iCloud photo albums.

**Comment by mikekenyon99** [comment]
I have fixed it... I found an album that was under the Shared Album section.... All good now apologies for logging ticket.

**Comment by RhetTbull** [comment]
If you want to exclude shared albums from the export you can use `--not-shared`
